cni/ipam/ipam_test.go

```diff
@@ -27,70 +30,191 @@ var ipamQueryResponse = "" +
 	"	</Interface>" +
 	"</Interfaces>"
 
-var localAsId string
-var poolId1 string
-var address1 string
+func TestIpam(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Ipam Suite")
+}
 
-// Wraps the test run with plugin setup and teardown.
-func TestMain(m *testing.M) {
-	var config common.PluginConfig
+// Handles queries from IPAM source.
+func handleIpamQuery(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(ipamQueryResponse))
+}
 
-	// Create a fake local agent to handle requests from IPAM plugin.
-	u, _ := url.Parse("tcp://" + ipamQueryUrl)
-	testAgent, err := common.NewListener(u)
+func parseResult(stdinData []byte) (*cniTypesCurr.Result, error) {
+	result := &cniTypesCurr.Result{}
+	err := json.Unmarshal(stdinData, result)
 	if err != nil {
-		fmt.Printf("Failed to create agent, err:%v.\n", err)
-		return
+		return nil, err
 	}
-	testAgent.AddHandler("/", handleIpamQuery)
+	return result, nil
+}
 
-	err = testAgent.Start(make(chan error, 1))
-	if err != nil {
-		fmt.Printf("Failed to start agent, err:%v.\n", err)
-		return
-	}
+func getStdinData(cniversion, ifname, subnet, ipAddress string) []byte {
+	stdinData := fmt.Sprintf(
+`{
+			"cniversion": "%s",
+			"master": "%s",
+			"ipam": {
+				"type": "internal",
+				"subnet": "%s",
+				"ipAddress": "%s"
+			}
+		}`, cniversion, ifname, subnet, ipAddress)
+	return []byte(stdinData)
+}
 
-	// Create the plugin.
-	plugin, err = NewPlugin("ipamtest", &config)
-	if err != nil {
-		fmt.Printf("Failed to create IPAM plugin, err:%v.\n", err)
-		return
-	}
+func getIfName() (string, error) {
 
-	// Configure test mode.
-	plugin.SetOption(common.OptEnvironment, common.OptEnvironmentAzure)
-	plugin.SetOption(common.OptAPIServerURL, "null")
-	plugin.SetOption(common.OptIpamQueryUrl, "http://"+ipamQueryUrl)
+	ifName := ""
 
-	// Start the plugin.
-	err = plugin.Start(&config)
+	interfaces, err := net.Interfaces()
 	if err != nil {
-		fmt.Printf("Failed to start IPAM plugin, err:%v.\n", err)
-		return
+		return ifName, err
 	}
 
-	// Run tests.
-	exitCode := m.Run()
-
-	// Cleanup.
-	plugin.Stop()
-	testAgent.Stop()
+	for _, iface := range interfaces {
+		ifName = iface.Name
+		break
+	}
 
-	os.Exit(exitCode)
+	return ifName, err
 }
 
-// Handles queries from IPAM source.
-func handleIpamQuery(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte(ipamQueryResponse))
-}
-
-//
-// CNI IPAM API compliance tests
-// https://github.com/containernetworking/cni/blob/master/SPEC.md
-//
-
-func TestAddSuccess(t *testing.T) {
-}
 
-func TestDelSuccess(t *testing.T) {
-}
+var (
+	_ = Describe("Test IPAM", func() {
+
+		var (
+			config common.PluginConfig
```

Comment:
looks like linter didn't work here

Reply:
I will move config to the Context where it works